interfaces/cos_agent/v0/README.md

```diff
@@ -0,0 +1,118 @@
+# `cos_agent`
+
+## Usage
+
+The `cos_agent` interface loads observability-related file paths and metrics endpoints into relation data.
+
+The first implementation was for the [`grafana-agent`](https://charmhub.io/grafana-agent/libraries/cos_agent) machine charm for "converting" paths to log files, alert rules and dashboards into relation data for the `loki_push_api`, `prometheus_remote_write` and `grafana_dashboards` interfaces.
+
+
+## Direction
+
+```mermaid
+flowchart TD
+    Provider["Provider (e.g. kafka)"] -- "relation_name
+    metrics_endpoints
+    metrics_rules_dir
+    logs_rules_dir
+    recursive_rules_dir
+    log_slots
+    dashboard_dirs
+    refresh_events" --> Requirer
+    Requirer["Requirer (e.g. grafana-agent)"] -- refresh_events --> Provider
+```
+
+As all Juju relations, the `cos_agent` interface consists of a provider and a requirer. In this case the `Provider` side of the relationship may provide telemetry settings, if not, the required side will use default values.
+
+## Behavior
+
+### Provider
+
+- Is expected to be able to provide metrics endpoints to be scraped.
+- Is expected to be able to provide a directory containing Prometheus style metrics rules files.
+- Is expected to be able to provide a directory containing Loki style rules files.
+- Is expected to be able to inform if metrics and logs rules directories should be scanned recursively.
+- Is expected to be able to provide snap slots to connect to for scraping logs.
+- Is expected to be able to provide a list of directories containing Grafana dashboard files.
+- Is expected to be able to provide a list of events on which to refresh relation data.
+
+
+### Requirer
+- Is expected to be able to scrape Prometheus metrics from metrics endpoints and remote-write these metrics to Prometheus and Prometheus compatible systems.
```

Comment:
I'm wondering if here we should be focusing on the behaviour rather than on the actor.
I.e. instead of "is expected to forward this data to Loki": "is expected to [do whatever loki would do]". The fact that we 'subcontract' the logic to another charm is not relevant. The requirer could in principle do all work itself and we wouldn't notice nor care.

Reply:
You have a solid point here. We should focus on the behaviour rather than the actor.
With that said, we can remove completely this section.